experimental/BasisLieHighestWeight/src/ModuleData.jl

```diff
@@ -0,0 +1,86 @@
+abstract type ModuleData end
+
+# To be implemented by subtypes:
+# Mandatory:
+#   dim(M::MyModuleData) -> ZZRingElem
+#   character(M::MyModuleData) -> Dict{WeightLatticeElem,<:IntegerUnion}
+
+
+mutable struct SimpleModuleData <: ModuleData
+    L::LieAlgebra
+    highest_weight::WeightLatticeElem
+
+    # The following fields are not set by default just set for caching
+    dim::ZZRingElem
+    character::Dict{WeightLatticeElem,<:IntegerUnion}
+
+    function SimpleModuleData(L::LieAlgebra, highest_weight::WeightLatticeElem)
+        new(L, highest_weight)
+    end
+end
+
+function L(M::SimpleModuleData)
+    return M.L
+end
+
+function highest_weight(M::SimpleModuleData)
+    return M.highest_weight
+end
+
+function dim(M::SimpleModuleData)
+    if !isdefined(M, :dim)
+        M.dim = dim_of_simple_module(ZZRingElem, M.L, M.highest_weight)
```

Comment:
```suggestion
        M.dim = dim_of_simple_module(ZZRingElem, base_lie_algebra(M), highest_weight(M))
```
better to use the getters where possible. Of course, for `.dim` it is not possible.
same at other places